src/database/filter/Filters.ts

```diff
@@ -0,0 +1,32 @@
+import { AmountLimitFilter } from './AmountLimitFilter.js';
+
+export class Filters {
+    signerToken: Record<string, AmountLimitFilter>
+    senderToken: Record<string, AmountLimitFilter>
+
+    constructor() {
+        this.signerToken = {}
+        this.senderToken = {}
+    }
+
+    addSignerToken(token: string, amount: number): void {
+        this.editTokenMap(this.signerToken, token, amount);
+    }
+
+    addSenderToken(token: string, amount: number): void {
+        this.editTokenMap(this.senderToken, token, amount);
+    }
+
+    private editTokenMap(map: Record<string, AmountLimitFilter>, token: string, amount: number): void {
```

Comment:
Add a check here if the amount is a positive number. Also token should not be case sensitive, so add a `toLowerCase()` here.

Reply:
Done !